etcdserver/v3demo_server.go

```diff
@@ -227,8 +282,16 @@ func applyUnion(kv dstorage.KV, union *pb.RequestUnion) *pb.ResponseUnion {
 	}
 }
 
-func applyCompare(kv dstorage.KV, c *pb.Compare) (int64, bool) {
-	ckvs, rev, err := kv.Range(c.Key, nil, 1, 0)
+// applyCompare applies the compare request.
+// applyCompare should only be called within a txn request and an valid txn ID must
+// be presented. Or applyCompare panics.
+// It returns the revision at which the comparison happens. If the comparison
+// succeeds, the it returns true. Otherwise it returns false.
+func applyCompare(txnID int64, kv dstorage.KV, c *pb.Compare) (int64, bool) {
+	if txnID == noTxn {
+		panic("applyCompare called with noTxn")
+	}
+	ckvs, rev, err := kv.TxnRange(txnID, c.Key, nil, 1, 0)
 	if err != nil {
```

Comment:
sorry. i cannot follow your thoughts. My previous sentence is in this scenario: if the txnID is a positive number, but it is not valid, it may deserve to panic here because the comparison doesn't happen actually.